### PR TITLE
converter: elaborate error messages @open sesame 3/28 19:04

### PR DESCRIPTION
**Pull request by myungjoo**
When there is a pipeline error, especially at the construct phase,
users need elaborated error messages that provide useful infomation
to correct pipeline descriptions.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 1]
```diff
@@ -664,6 +664,38 @@ gst_tensors_info_get_names_string (const GstTensorsInfo * info)
...
+ * @brief GstTensorsInfo represented as a string. Caller should free it.
+ * @param info GstTensorsInfo structure
+ * @return The newly allocated string representing the tensorsinfo. Free after use.
+ */
+gchar *
+gst_tensors_info_to_string (const GstTensorsInfo * info)
+{
+  GString *gstr = g_string_new (NULL);
+  unsigned int i;
+  int limit = info->num_tensors;
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:677]: type would be unsigned int

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 2]
```diff
@@ -664,6 +664,38 @@ gst_tensors_info_get_names_string (const GstTensorsInfo * info)
...
+
+  g_string_append_printf (gstr, "Num_Tensors = %u, Tensors = [",
+      info->num_tensors);
+  if (limit > NNS_TENSOR_SIZE_LIMIT) {
+    limit = NNS_TENSOR_SIZE_LIMIT;
+    g_string_append_printf (gstr,
+        "(Num_Tensors out of bound. Showing %d only)", limit);
+  }
+  for (i = 0; i < info->num_tensors; i++) {
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:686]: need change `for (i = 0; i < info->num_tensors; i++)` >> `for (i = 0; i < limit; i++)`

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 3]
```diff
@@ -664,6 +664,38 @@ gst_tensors_info_get_names_string (const GstTensorsInfo * info)
...
+    const char *name = info->info[i].name;
+    const gchar *type = gst_tensor_get_type_string (info->info[i].type);
+    gchar *dim = gst_tensor_get_dimension_string (info->info[i].dimension);
+
+    g_string_append_printf (gstr, "{\"%s\", %s, %s}%s",
+        name, type, dim, (i == info->num_tensors - 1) ? "]" : ", ");
+
+    g_free (dim);
+  }
+  return gstr->str;
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:696]: `return g_string_free (gstr, FALSE)` to release GString.

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 4]
```diff
@@ -779,6 +811,26 @@ gst_tensors_config_copy (GstTensorsConfig * dest, const GstTensorsConfig * src)
...
+  GString *gstr = g_string_new (NULL);
+  const gchar *fmt = gst_tensor_get_format_string (config->format);
+  g_string_append_printf (gstr, "Format = %s, Framerate = %d/%d",
+      fmt, config->rate_n, config->rate_d);
+  if (config->format == _NNS_TENSOR_FORMAT_STATIC) {
+    gchar *infostr = gst_tensors_info_to_string (&config->info);
+    g_string_append_printf (gstr, ", %s", infostr);
+    g_free (infostr);
+  }
+  return gstr->str;
```
Comment (jaeyun-jung) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:831]: Also, need change `return g_string_free (gstr, FALSE)`

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c (hunk 1 above)
Comment (myungjoo) [reply gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:677]: applied.

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c (hunk 2 above)
Comment (myungjoo) [reply gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:686]: applied.

**Review by jaeyun-jung — APPROVED** [review]

**Review by anyj0527 — APPROVED** [review]
Those two `..._to_string` function would be helpful greatly. LGTM 👍

**Review by songgot — APPROVED** [review]
LGTM

**Pull request merged**